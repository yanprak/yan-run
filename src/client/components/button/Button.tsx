import React, { FC } from 'react';
import { OwnProps } from './types';
import './button.scss';

const Button: FC<OwnProps> = (props: OwnProps) => {
  const {
    children,
    size,
    styleType,
    ...otherProps
  } = props;
  const className = `button button_${styleType} button_${size} ${props.className ? props.className : ''}`;
  return (
    <button type="button" {...otherProps} className={className}>{children}</button>
  );
};

export default Button;
