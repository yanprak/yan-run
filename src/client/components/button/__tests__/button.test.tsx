import React from 'react';
import TestRenderer from 'react-test-renderer';
import Button from '../Button';

test('button test', () => {
  const component = TestRenderer.create(
    <Button size="small" styleType="secondary">Кнопка</Button>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
