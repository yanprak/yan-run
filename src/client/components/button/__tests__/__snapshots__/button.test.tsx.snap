// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test 1`] = `
<button
  className="button button_secondary button_small "
  type="button"
>
  Кнопка
</button>
`;
